flink-table/flink-table-planner/src/main/java/org/apache/flink/table/planner/operations/converters/SqlNodeConverters.java

```diff
@@ -37,9 +40,9 @@ public class SqlNodeConverters {
     static {
         // register all the converters here
         register(new SqlCreateCatalogConverter());
-        register(new SqlCreateTableAsConverter());
-        register(new SqlCreateTableLikeConverter());
-        register(new SqlCreateTableConverter());
+        register(new SqlCreateTableConverter(), TypeInformation.of(SqlCreateTable.class));
+        register(new SqlCreateTableConverter(), TypeInformation.of(SqlCreateTableAs.class));
+        register(new SqlCreateTableConverter(), TypeInformation.of(SqlCreateTableLike.class));
```

Comment:
IMO, this breaks the design of `SqlNodeConverter` somehow. The principle of `SqlNodeConverter` focuses on a single kind of SqlNode conversion, but this involves `if else` and `instance of` which we would like to avoid in `SqlNodeToOperationConversion`.

Is there any problems to continue the way of 3 separate converters?

Reply:
Yes, there will be the following situation:
If I only register one SqlCreateTable converter, other subclasses will not be able to obtain the instance of the SqlCreateTable converter, because when registering, I get the SqlCreateTable class as the Key through type analysis, and for subclasses: such as SqlCreateTableAs, it is Use the SqlCreateTableAs type as a key to find the converter, so it is not available.

Reply:
Or create separate SqlNodeConverter for SqlCreateTableAs and SqlCreateTableLike.

Reply:
Yes, that what I mean "3 separate converters" and that is how you did in the first commit.

Reply:
I have revised it back.